js-miniapp-sdk/src/types/responseTypes/interstitial/index.ts

```diff
@@ -0,0 +1,9 @@
+import { AdTypes } from '../../adTypes';
+import { AdResponse } from '../AdResponse';
+/**
+ * Intterstitial ad type response from native sdk
+ */
+interface InterStitial {
+  adType: AdTypes.INTERSTITIAL;
+}
+export type InterstitialAdResponse = AdResponse & InterStitial;
```

Comment:
Why not extend `AdResponse` instead of using an intersection type?

```
interface InterstitialAdResponse extends AdResponse {
  adType: AdTypes.INTERSTITIAL;
}
```

It seems like extension better fits our use case because `InterstitialAdResponse` IS an `AdResponse`. Also, it seems like using the intersection type causes our generated documentation to be a little strange: there's both an `InterstitialAdResponse` type and a `InterstitialAd` interface, but neither of them shows the whole type because you have to click on `AdResponse` to see it's properties. Using extension would solve this issue.

Reply:
Extended!